### ISSUE TITLE
[java] AccessorMethodGeneration false positive with overloads

### ISSUE THREAD

**Issue by TWiStErRob** [pmd/pmd#807]
**Affects PMD Version:** 5.8.1, 6.0.0 (and probably 5.5.4 upwards since the rule was introduced)

**Rule:** `AccessorMethodGeneration`

**Description:** overloaded method tricks the rule into thinking there will be an accessor method generated.

**Code Sample demonstrating the issue:**

```java
@SuppressWarnings({"unused", "StaticMethodOnlyUsedInOneClass", "SameParameterValue", "PMD.UseUtilityClass"})
class Outer {

  static void outerMethod(int i) {
    // package private method, safe to call, and is called from Inner.f()
  }

  private static void outerMethod(String s) {
    // private method, not used anywhere, yet triggers AccessorMethodGeneration
  }

  static class Inner {
    void f() {
      outerMethod(0);
    }
  }
}
```

**Running PMD through:** Gradle (5.8.1) and CLI (6.0.0 `pmd -d . -format text -R java-design`)

**Comment by jsotuyod** [comment]
@TWiStErRob thanks for the report. This is closely related to #342. Once we start merging symbol table data with type resolution we should be able to fix it.

**Comment by TWiStErRob** [comment]
I was that, but it was related to fields, so I reported this, but now I see that #613 is very similar.
What's the result of GSoC 2017?

**Comment by jsotuyod** [comment]
@TWiStErRob over GSoC 2017 type resolution got a really impressive enhancement. 

* 5.8.0: support for `this` / `super`. Support for fields, even when chained. Method and fields inheritance supported.
* 6.0.0: added support for method shadowing / overloading and varargs when not using generics. The support for generics is partial (we have not completed the implementation of our type inference engine, still plenty of TODOs to fill). Static fields are supported too. Type resolution based implementation of rules are promoted to default replacing the old ones.

The type resolution data is now complete enough to be useful, but we would still miss more cumbersome scenarios. We look forward for volunteers to help us fill the gaps.

**Comment by adangel** [comment]
Fixed via #2899 for PMD 7.
